README.md

```diff
@@ -1,178 +1,114 @@
-    endpoint: "localhost:3979"
-}
-```
+**note:** a goal to consider is the service credentials being *templated* in this file. we've made assumptions in the past about where these credentials should be stored and their shape.
 
 ## Composer configuration
 
 ### config.json
 
-config.json in composer is where to specfic 
-1. which bot we are editing (if we follow vscode convention, this should be in .composer, but let's make it simple here)
-2. which launcher we are using
-
 ```
-/Composer/config.json 
 {
-    // this is the bot we are editing\composing
-    bot: {
-        provider: "localDisk"
-        path: "../../Bots/SampleBot1/bot1.bot"
-    }
-    
-    launcher: {
-        startCommand: "cp launcher.json '../../BotLauncher/CSharp/appsetings.json' && cd '../../BotLauncher/CSharp' && dotnet run"
+  // the path to the BotLauncher root
+  launcherDir: "../../BotLauncher",
+  bots: [
+    //an array of bots to load in this application
+    {
+      path: "../../Bots/SampleBot1/bot1.bot",
+      secrets: "../path/to/appsettings.json" // the path to credentials necessary to run the bot (and template into the .bot file)
+    },
+    {
+      path: "../../Bots/SampleBot2/bot2.bot",
+      secrets: "../path/to/local.env"
     }
+  ]
 }
 ```
 
-It's very self-explained, the "bot" section tell the composer where the bot is, the "launcher" section tell the composer how to play with the launcher. 
-
-### launcher4[CS\NODE].json
-
-there is also a `launcher4CS.json` inside composer, which is designed to override the config inside launcher, to make sure the launcher is started the way composer wants.
-
-``` 
-/Composer/launcher4CS.json (same schema as /BotLauncher/CSharp/appsettings.json)
-
-this whole file will be copied to the Bot Launcher folder to start Launcher
-
-{
-    bot: {
-        provider: "localDisk",
-        path: "C://Bots/SampleBot1/bot1.bot"  // note there the value is faltten, become this is supposed to copied into 
-    }
-
-    // it may include some othe info to run like
-    endpoint: "localhost:3979"
-}
-
-```
-So, if you look at the startCommand of launcher in package.json (insider composer), you will see this launcher.json is copied to the launcher then use it to start. 
-
-An alertative way to this "copy and start" is to tell the launcher config file using command line such as
-```
-  dotnet run -f launcher4CS.json
-```
-
-this may also make sense. but the way we see the benifits of the copy and start approach can help make sure the configuration is complete in the launcher folder, thus it won't block the later deployment step. 
-
-## conclusion
-that's it, the high-level architecture is simple, i guess. 
-
+To start the bot with updated dialogs the application references the "start" property in the .bot file and packs up up as a command line argument sent to the BotLauncher. The BotLauncher is able to infer the runtime (dotnet or node).
 
+example: `node index.js --bot=${JSON.stringify(botFile)}`
```

Comment:
an example of passing the botfile information to the BotLauncher. would something like this work?

Reply:
this would work if the bot launcher knows how to handle that, which means bot launcher is designed in this way, i'm totally ok with this approach, we don't have to copy\paste file around